invenio_rdm_records/services/access/service.py

```diff
@@ -506,97 +533,25 @@ def read_all_grants(self, identity, id_, expand=False):
         # Permissions
         self.require_permission(identity, "manage", record=record)
 
-        # Fetching
-        return self.grants_result_list(
-            service=self,
-            identity=identity,
-            results=parent.access.grants,
-            expand=expand,
-        )
-
-    def read_all_grants_by_subject(self, identity, id_, subject_type, expand=False):
-        """Read access grants of a record (resp. its parent) by subject type."""
-        record, parent = self.get_parent_and_record_or_draft(id_)
+        existing_grants = parent.access.grants
 
-        # Permissions
-        self.require_permission(identity, "manage", record=record)
-
-        user_grants = []
-        for grant in parent.access.grants:
-            if grant.subject_type == subject_type:
-                user_grants.append(grant)
+        for grant in existing_grants:
+            # removes group grants if groups are not enabled in the instance
+            if grant.subject_type == this.group_subject_type:
+                try:
+                    self.require_permission(
+                        identity,
+                        "groups_enabled",
+                        member_types={RecordAccessService.group_subject_type},
+                    )
+                except PermissionDeniedError:
+                    existing_grants.remove(grant)
 
         # Fetching
         return self.grants_result_list(
             service=self,
             identity=identity,
-            results=user_grants,
-            expand=expand,
-        )
-
-    @unit_of_work()
-    def update_grant_by_subject(
```

Comment:
The victim of rebase of previous PRs (methods `update_grant_by_subject` and `read_all_grants_by_subject` were duplicated)